page-tracking/src/page_info.rs

```diff
@@ -171,6 +170,10 @@ impl PageInfo {
     /// Pops the current owner if there is one, returning the page to the previous owner.
     pub fn release(&mut self) -> PageTrackingResult<PageOwnerId> {
         use PageState::*;
+        // Converted and locked pages cannot be released
+        if matches!(self.state, Converted) && self.is_locked() {
```

Comment:
I would moved the `is_locked()` check to the `Mapped | VmState | ...` match arm below.